doc/proposals/sdk-integration-with-olm.md

```diff
@@ -0,0 +1,212 @@
+* Operator developers can use `operator-sdk` to build a catalog/bundle containing their Operator for use with OLM
+
+#### Specific
+
+* `operator-sdk` creates a [bundle][bundle] from an Operator project to deploy with OLM
+* `operator-sdk` has a CLI interface to interact with OLM
+* `operator-sdk` installs a specific version of OLM onto Kubernetes cluster
+* `operator-sdk` uninstalls a specific version of OLM onto Kubernetes cluster
+* `operator-sdk` accepts a bundle and deploys that operator onto an OLM-enabled Kubernetes cluster
+* `operator-sdk` accepts a bundle and removes that operator onto an OLM-enabled Kubernetes cluster
+
+### Non-Goals
+
+## Proposal
+
+### User Stories
+
+**TODO**
+
+Detail the things that people will be able to do if this is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+
+#### Story 1
+
+### Implementation Details/Notes/Constraints
+
+Initial PR: https://github.com/operator-framework/operator-sdk/pull/1912
+
+#### Use of operator-registry
+
+The SDK's approach to deployment should be as general and reliant on existing mechanisms as possible. To that end, [`operator-registry`][registry] should be used since it defines what a bundle contains and how one is structured. `operator-registry` libraries should be used to create and serve bundles, and interact with package manifests.
+
+The idea is to create a `Deployment` containing the latest `operator-registry` [image][registry-image] to initialize a bundle database and run a registry server serving that database using binaries contained in the image. The `Deployment` will contain volume mounts from a `ConfigMap` containing bundle files and a package manifest for an operator. Using manifest data in the `ConfigMap` volume source, the registry initializer can build a local database and serve that database through the `Service`. OLM-specific resources created by the SDK or supplied by a user, described below, will establish communication between this registry server and OLM.
+
+#### OLM resources
+
+OLM understands `operator-registry` servers and served data through several objects. A [`CatalogSource`][olm-catalogsource] specifies how to communicate with a registry server. A [`Subscription`][olm-subscription] links a particular CSV channel to a `CatalogSource`, indicating from which `CatalogSource` OLM should pull an Operator. Another OLM resource that _may_ be required is an [`OperatorGroup`][olm-operatorgroup], which provides Operator namespacing information to OLM; OLM creates two `OperatorGroup`'s by default, one of which can be used for globally scoped Operators.
+
+These resources can be created from bundle data with minimal user input. They can also be created from manifests defined by the user; however, the SDK cannot make guarantees that user-defined manifests will work as expected.
+
```

Comment:
I think we could specify here what resources we are speaking about and link its [docs](https://github.com/operator-framework/community-operators/blob/master/docs/testing-operators.md#1-create-the-operatorsource) as where it would be created as well. E.g here in my project I created the path `<project-name>/deploy/olm-catalog/olm-test` since it can be used just to test it locally. See [here](https://github.com/dev4devs-com/postgresql-operator/tree/master/deploy/olm-catalog/olm-test)